src/Numerics/Mesh/BrickMesh.jl

```diff
@@ -398,30 +402,25 @@ function parallelsortcolumns(
     ]
 
     cols = map(i -> view(A, :, i), 1:n)
-    sendstarts = [
-        (i <= csize) ?
+    senddispls = Cint[
         (
             searchsortedfirst(cols, pivots[:, i], lt = lt, by = by, rev = rev) - 1
-        ) * m + 1 :
-        n * m + 1 for i in 1:(csize + 1)
+        ) * m for i in 1:csize
+    ]
+    sendcounts = Cint[
+        (i == csize ? n * m : senddispls[i + 1]) - senddispls[i]
+        for i in 1:csize
     ]
-    sendcounts = [Cint(sendstarts[i + 1] - sendstarts[i]) for i in 1:csize]
 
-    B = []
-    for r in 0:(csize - 1)
-        counts = MPI.Allgather(sendcounts[r + 1], comm)
-        c = MPI.Gatherv(
-            view(A, sendstarts[r + 1]:(sendstarts[r + 2] - 1)),
-            counts,
-            r,
-            comm,
-        )
-        if r == crank
-            B = c
-        end
-    end
+    recvcounts = similar(sendcounts)
+    MPI.Alltoall!(UBuffer(sendcounts, 1), MPI.UBuffer(recvcounts, 1), comm)
+    B = similar(A, sum(recvcounts))
+    MPI.Alltoallv!(
+        VBuffer(A, sendcounts, senddispls),
+        VBuffer(B, recvcounts),
+        comm,
+    )
```

Comment:
@lcw @jkozdon Is there any particular reason to prefer a loop of `Gatherv`s instead of an `Alltoallv`?

Reply:
My guess is it's @lcw proclivity to avoid collectives since he's been burned by them in the past (for instance an all reduce not working on a machine when the number of ranks was large enough). My understanding is this is particularly a problem with new, cutting edge machines when the number of MPI ranks is pushing the boundary of what distributions have handled.

Reply:
Not that I know of.  Definitely cleaner with the `Alltoallv`.